flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/GenericHiveMetastoreCatalog.java

```diff
@@ -363,4 +333,23 @@ public CatalogColumnStatistics getPartitionColumnStatistics(ObjectPath tablePath
 		throw new UnsupportedOperationException();
 	}
 
+	// ------ utils ------
+
+	/**
+	 * Filter out Hive-created properties, and return Flink-created properties.
+	 */
+	private Map<String, String> retrieveFlinkProperties(Map<String, String> hiveTableParams) {
+		return hiveTableParams.entrySet().stream()
+			.filter(e -> e.getKey().startsWith(FLINK_PROPERTY_PREFIX))
+			.collect(Collectors.toMap(e -> e.getKey().replace(FLINK_PROPERTY_PREFIX, ""), e -> e.getValue()));
+	}
+
+	/**
+	 * Add a prefix to Flink-created properties to distinguish them from Hive-created properties.
+	 */
+	private Map<String, String> buildFlinkProperties(Map<String, String> properties) {
```

Comment:
Maybe we can use words "mask" and "unmask" instead of "build" and "retrieve".

Reply:
I feel "mask" is a good name. Not sure about "unmask" because the retrieval not only unmasks flink properties but also filters out hive properties.